### PR TITLE
Unzip fix wrong password supplied

### PR DESCRIPTION
**Pull request by jochman**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/20555

## Description
Checking for a wrong password supplied instead of just pushing a file.

## Screenshots
image

## Does it break backward compatibility?
   - No

## Must have
- [X] Tests
- [X] Documentation (with link to it)
- [ ] Code Review

- [ ] [CHANGELOG](https://github.com/demisto/content/compare/unzip-fix-empty-zip?expand=1#diff-e261ed4f4532b6d97d3338288945aada)

### CODE REVIEW

File: Scripts/UnzipFile/UnzipFile.py [hunk 1]
```diff
@@ -7,109 +7,110 @@
...
+    password = args.get('password', None)
+
+    filenames = []
+    # remembering which files and dirs we currently have so we add them later as newly extracted files.
+    excluded_files = [f for f in os.listdir('.') if isfile(f)]
+    excluded_dirs = [d for d in os.listdir('.') if isdir(d)]
+    # extracting the zip file
+    process = Popen(["7z", "x", "-p{}".format(password), file_path], stdout=PIPE, stderr=PIPE)
+    stdout, stderr = process.communicate()
+    if "Wrong password?" in stdout:
```
Comment (Itay4) [inline Scripts/UnzipFile/UnzipFile.py]:
```suggestion
    if 'Wrong password?' in stdout:
```

File: Scripts/UnzipFile/UnzipFile.py [hunk 2]
```diff
@@ -7,109 +7,110 @@
...
+
+    password = args.get('password', None)
+
+    filenames = []
+    # remembering which files and dirs we currently have so we add them later as newly extracted files.
+    excluded_files = [f for f in os.listdir('.') if isfile(f)]
+    excluded_dirs = [d for d in os.listdir('.') if isdir(d)]
+    # extracting the zip file
+    process = Popen(["7z", "x", "-p{}".format(password), file_path], stdout=PIPE, stderr=PIPE)
+    stdout, stderr = process.communicate()
```
Comment (Itay4) [inline Scripts/UnzipFile/UnzipFile.py]: also check stderr

File: Scripts/UnzipFile/UnzipFile.py [hunk 3]
```diff
@@ -7,109 +7,111 @@
...
+    if stderr:
+        return_error(stderr)
```
Comment (Itay4) [inline Scripts/UnzipFile/UnzipFile.py]:
```suggestion
        return_error(str(stderr))
```

**Comment by Itay4** [comment]
@roysagi @jochman whats the deal here with the changelog?

**Comment by jochman** [comment]
@roysagi have the permission to review changelog entries.

**Pull request merged**